src/functions/genfrac.js

```diff
@@ -321,3 +335,137 @@ defineFunction({
+
+        // Look into the parse nodes to get the desired delimiters.
+        let leftNode = checkNodeType(args[0], "ordgroup");
+        if (leftNode) {
+            leftNode = assertNodeType(leftNode.value[0], "open");
+        } else {
+            leftNode = assertNodeType(args[0], "open");
+        }
+        const leftDelim = delimFromValue(leftNode.value);
+
+        let rightNode = checkNodeType(args[1], "ordgroup");
+        if (rightNode) {
+            rightNode = assertNodeType(rightNode.value[0], "close");
+        } else {
+            rightNode = assertNodeType(args[1], "close");
+        }
+        const rightDelim = delimFromValue(rightNode.value);
+
+        const barNode = assertNodeType(args[2], "size");
+        let hasBarLine;
+        let barSize = null;
+        if (barNode.value.isBlank) {
+            // \genfrac acts differently than \above.
+            // \genfrac treats an empty size group as a signal to use a
+            // standard bar size. \above would see size = 0 and omit the bar.
+            hasBarLine = true;
+        } else {
+            barSize = barNode.value.value;
+            hasBarLine = barSize.number > 0;
+        }
+
+        // Find out if we want displaystyle, textstyle, etc.
+        let size = "auto";
+        let styl = checkNodeType(args[3], "ordgroup");
+        if (styl) {
+            if (styl.value.length > 0) {
+                size = stylArray[Number(styl.value[0].value)];
+            }
+        } else {
+            styl = assertNodeType(args[3], "textord");
+            size = stylArray[Number(styl.value)];
+        }
+
+        return new ParseNode("genfrac", {
+            type: "genfrac",
+            numer: numer,
+            denom: denom,
+            continued: false,
+            hasBarLine: hasBarLine,
+            barSize: barSize,
+            leftDelim: leftDelim,
+            rightDelim: rightDelim,
+            size: size,
+        }, parser.mode);
+    },
+
+    htmlBuilder,
+    mathmlBuilder,
+});
+
+// \above is an infix fraction that also defines a fraction bar size.
+defineFunction({
+    type: "infix",
+    names: ["\\above"],
+    props: {
+        numArgs: 1,
+        argTypes: ["size"],
+        infix: true,
+    },
+    handler({parser, funcName, token}, args) {
+        const sizeNode = assertNodeType(args[0], "size");
+        return new ParseNode("infix", {
+            type: "infix",
+            replaceWith: "\\\\abovefrac",
+            sizeNode: sizeNode,
+            token: token,
+        }, parser.mode);
+    },
+});
+
+defineFunction({
+    type: "genfrac",
+    names: ["\\\\abovefrac"],
+    props: {
+        numArgs: 3,
+        argTypes: ["math", "size", "math"],
+    },
+    handler: ({parser, funcName}, args) => {
+        const numer = args[0];
+        const sizeNode = assertNodeType(args[1], "infix");
+        const denom = args[2];
+
+        // $FlowFixMe
+        const barSize = sizeNode.value.sizeNode.value.value;
```

Comment:
The following should work:
```
const infixNode = assertNodeType(args[1], "infix");
const sizeNode = assertNodeType(infixNode.value.sizeNode);
const barSize = sizeNode.value.value;
```
@marcianx does that look right?

Reply:
Looks right, except you were missing the "size" in the second line:
```
const sizeNode = assertNodeType(infixNode.value.sizeNode, "size");
```
by any chance?

Reply:
Done. Thank you for the help.